lib/ruby_lsp/requests/diagnostics.rb

```diff
@@ -30,15 +30,20 @@ def initialize(uri, document)
 
       sig do
         override.returns(
-          T.any(
-            T.all(T::Array[Support::RuboCopDiagnostic], Object),
-            T.all(T::Array[Support::SyntaxErrorDiagnostic], Object),
+          T.nilable(
+            T.any(
+              T.all(T::Array[Support::RuboCopDiagnostic], Object),
+              T.all(T::Array[Support::SyntaxErrorDiagnostic], Object),
+            ),
           ),
         )
       end
       def run
-        return [] if @document.syntax_error?
-        return [] unless defined?(Support::RuboCopDiagnosticsRunner)
+        return if @document.syntax_error?
+        return unless defined?(Support::RuboCopDiagnosticsRunner)
+
+        # Don't try to run RuboCop diagnostics for files outside the current working directory
+        return unless @uri.sub("file://", "").start_with?(Dir.pwd)
```

Comment:
What was your thinking around returning nils instead of empty arrays as we did previously?

Reply:
This is just my preference, but I think returning `nil` better conveys that there are no diagnostics for this file rather than an empty array.

Reply:
I support this decision